### PR TITLE
issue 48

### PR DESCRIPTION
**Pull request by nguyenvo8**
Removed existingSettingsFile() in DataController class.
Removed StudyPlanner and then added it back for testing purposes.
Changed the behavior of validateNumber in AccountController so it will turn red when the user inputs invalid input.

### CODE REVIEW

**Comment by nguyenvo8** [comment]
I closed the first pull request because I forgot to stage all the file, so I made another pull request after staged the files

**Comment by rsanchez-wsu** [comment]
@nguyenvo8 Please note that it is not correct to close a pull request and open a new one in this manner.  The correct approach is to fix whatever problems, push the new commit(s), and let GitHub update the pull request.  That way development history is preserved.

File: src/Controller/AccountController.java [hunk 1]
```diff
@@ -130,6 +130,7 @@ public void validateNumber()
         if (account_no.getText().trim().isEmpty())
             this.submit.setDisable(true);
         else
+            this.account_no.setStyle("-fx-text-box-border:red;");
```
Comment (rsanchez-wsu) [inline src/Controller/AccountController.java]: This is a bug.  You have added a line to the else which now results in the next line (`this.handleChange();`) being executed unconditionally.  You need wrap the two statements in curly braces.

File: src/Controller/DataController.java [hunk 2]
```diff
@@ -107,6 +94,7 @@ static private HubFile processUpdateHubFile(NodeList nList) throws Exception
                                 break;
                             case "module":
                                 // not yet added
+
```
Comment (rsanchez-wsu) [inline src/Controller/DataController.java]: This line seems to have been added inadvertently.

File: src/Controller/DataController.java [hunk 3]
```diff
@@ -129,6 +117,8 @@ static private HubFile processUpdateHubFile(NodeList nList) throws Exception
                             case "examEvent":
                                 // not yet added
                                 break;
+                        default:
```
Comment (rsanchez-wsu) [inline src/Controller/DataController.java]: What is the purpose of adding this default clause?  Should it do something (like throw an exception or log an error)?

File: src/Controller/DataController.java [hunk 4]
```diff
@@ -388,6 +378,8 @@ static private HubFile processNewHubFile(NodeList nList) throws Exception
                                     }
 
                                     break;
+                            default:
```
Comment (rsanchez-wsu) [inline src/Controller/DataController.java]: What is the purpose of adding this default clause? Should it do something (like throw an exception or log an error)?

File: src/Controller/TaskController.java [hunk 5]
```diff
@@ -480,7 +480,7 @@ protected void updateItem(final Task item, final boolean empty)
         TableColumn<Task, String> nameColumn = new TableColumn<>("Task");
         nameColumn.setCellValueFactory(new PropertyValueFactory<>("name"));
 
-        TableColumn<Task, String> assignmentColumn = new TableColumn<>("Assignments");
+        TableColumn<Task, String> assignmentColumn = new TableColumn<Task,String>("Assignments");
```
Comment (rsanchez-wsu) [inline src/Controller/TaskController.java]: This change does not seem correct.  The Java compiler will properly infer the type argument of the `TableColumn` constructor.  There is no need to add an explicit type specification here.

**Comment by rsanchez-wsu** [comment]
@nguyenvo8  I don't normally like to close a pull request and a open a new one, but you have a commit (e841ff9) that includes numerous Eclipse project files which refer to user-specific and/or system-specific directories which will not be present on systems other than yours.  Additionally, the message for that commit indicates changes to DataController and StudyPlannerController, but no such changes are present.

I have also made specific individual comments on commit 764865d.  Please have a look at them and make the necessary changes before proceeding.

Please close this pull request, clean up the commit history (you may need to do a force push, which is something else I do not recommend but which may be necessary here), and open a new pull request. In particular, before you submit the pull request, please review the aggregate diff of the the changes of all the commits you are submitting via the pull request to make sure that only changes you specifically intend are included.  Finally, make sure that the commit message refers to the issue which is closed by it.

**Pull request closed without merging**